### PR TITLE
Display possibly related issues on the 'New Issue' page

### LINKED ISSUE #1138: Search issues while a new issue is being typed
Exactly like what stackoverflow does when creating a new question:
* as soon as the title has a few words
* start a somewhat fuzzy search query in the background
* display the results in the space above the title of the issue
* continue updating results while issue is being typed

Maybe limit to 2-5 search requests per minute so as not to eat all the user's rate limiting quota.

[Here's a video of someone doing a SO submission.](https://www.youtube.com/watch?v=OmyqGI91ZBU)

### PR DESCRIPTION
**Pull request by shroudedcode**
*A basic implementation of the feature suggested by @fenollp in #1138.*

When the title input field on the `New Issue` page is blurred, an API request is made to fetch issues on the repository with the title as the query. The first five issues are then displayed in the sidebar.

If there are more than five, an *X more results* link that links to the regular issue search is added below.

Currently, the search is really basic. While StackOverflow's suggestions get more accurate as you enter more text, this search only looks for issues that contain the text in the title which is not the best way to determine whether an issue is relevant. Please let me know if you have any ideas on how the search could be improved.

###### Closes #1138

### CODE REVIEW

**Comment by bfred-it** [comment]
We should also skip short queries (length < 4) and any *glue* words like “a”, “the”, “and” which could limit results unnecessarily.

**Comment by hkdobrev** [comment]
Could we show below "Helpful resources" so there's no jump?

**Comment by bfred-it** [comment]
@hkdobrev I tried placing it at the end of the sidebar but it's too far:

**Comment by sindresorhus** [comment]
This is truly awesome, @shroudedcode! I can't believe this is not a native GitHub feature.

**Comment by hkdobrev** [comment]
> I tried placing it at the end of the sidebar but it's too far

When you create an issue on a repo you don't have contribution access to, you don't need the sidebar items like the assignee, label, milestone and project on the new issue form as you don't have access to set them.

**Comment by hkdobrev** [comment]
Thanks @shroudedcode for doing that! 🎉

**Comment by bfred-it** [comment]
> When you create an issue on a repo you don't have contribution access to, you don't need the sidebar items like the assignee, reviewers, label, milestone and project on the new issue form as you don't have access to set them.

If you don't have contribution access, your sidebar is empty, so position matters only if you're a collaborator :D 

empty sidebar

File: test/searchable-words.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+test('Searchable words filter', t => {
+	t.deepEqual(fn(' n o t h i n g	n e s s'), []);
+	t.deepEqual(fn('Hello, world!'), ['Hello', 'world']);
+	t.deepEqual(fn(' 400  	[error] is\\ nonsense !'), ['400', 'error', 'nonsense']);
+	t.deepEqual(fn('1col is broken'), ['1col', 'broken']);
+	t.deepEqual(fn('ava is 🔥🔥🔥'), ['ava']);
+	t.deepEqual(fn('1身 分'), ['1身', '分']);
+	t.deepEqual(fn('làm 7/2 nộp'), ['làm', 'nộp']);
+	t.deepEqual(fn('测试'), ['测试']);
+	t.deepEqual(fn('حمید'), ['حمید']);
```
Comment (merwok) [inline test/searchable-words.js]: Stray brackets instead of parens here

Comment (bfred-it) [reply test/searchable-words.js]: Oh… how did this test pass??

Comment (bfred-it) [reply test/searchable-words.js]: hahahahahahahahah oh Allah.

Arabic is a RTL language and GitHub's syntax highlighter can't handle/contain the RTL change. Copy and paste the code into your editor and you'll see the correct order (works with Sublime)

**Comment by anandthakker** [comment]
👏  thanks @shroudedcode!

@sindresorhus what do you think about including a related features list when _reading_ an issue, also?  Could be useful for maintainers to quickly find duplicates (or just related issues that may be relevant to discussion)

**Comment by sindresorhus** [comment]
@anandthakker Possibly. Can you open a new issue about it? I think we should let this feature mature for a bit first though. I'm sure it could be smarter, and getting false-positives is a lot more annoying when reading issues than when writing.

**Comment by varkor** [comment]
I find that recently (presumably since this landed), typing in the title area is not very responsive. Is this being sufficiently throttled (and is anyone else having this problem)?

**Comment by bfred-it** [comment]
Definitely not the case; it does not update while typing but only when you leave the field (`blur` event)

**Comment by varkor** [comment]
Ah, must be something else then, thanks!

**Pull request merged**